flaml/automl.py

```diff
@@ -1552,14 +1565,19 @@ def custom_metric(
                 the searched learners, such as sample_weight. Include period as
                 a key word argument for 'ts_forecast' task.
         """
-        from .nlp.utils import _is_nlp_task
 
         self._state._start_time_flag = self._start_time_flag = time.time()
         if task == FORECAST:
             self._state.task = TS_FORECAST
         else:
             self._state.task = task
         self._state.log_training_metric = log_training_metric
+
+        from .nlp.utils import _is_nlp_task
+
+        if _is_nlp_task(task):
+            fit_kwargs["is_retrain"] = False
+
```

Comment:
Don't specialize for nlp task here. You can check the 'earlystop' branch to get a sense how to set the retrain flag.